subsys/bluetooth/scan.c

```diff
@@ -1153,6 +1356,8 @@ void bt_scan_init(const struct bt_scan_init_param *init)
 
 		bt_scan.connect_if_match = false;
 	}
+
+	bt_conn_cb_register(&conn_callbacks);
```

Comment:
We could register the callbacks only if the `CONFIG_BT_SCAN_CONN_ATTEMPTS_FILTER` is enabled. We could save some RAM if we do not use the connection attempts filter.

Reply:
Good point. Done